pkg/internal/discover/finder.go

```diff
@@ -98,5 +98,9 @@ func newGoTracersGroup(cfg *beyla.Config, metrics imetrics.Reporter) []ebpf.Trac
 }
 
 func newGenericTracersGroup(cfg *beyla.Config, metrics imetrics.Reporter) []ebpf.Tracer {
-	return []ebpf.Tracer{generictracer.New(cfg, metrics), tctracer.New(cfg)}
+	if cfg.EBPF.UseLinuxTC {
+		return []ebpf.Tracer{generictracer.New(cfg, metrics), tctracer.New(cfg)}
```

Comment:
This needs to be done here, because simply not calling `tctracer.SetupTC()` is not enough. The mere attempt to load the ebpf program (as in `ebpf.LoadAndAssign()`), which happens way before `SetupTC()` is called, will trigger the kernel verifier.

This also keeps the logic giving it visibility where it belongs IMHO.